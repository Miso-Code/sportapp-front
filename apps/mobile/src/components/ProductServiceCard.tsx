import React from 'react'
import { useTranslation } from 'react-i18next'
import { View, StyleSheet } from 'react-native'
import { Card, Text, Chip, MD3Theme, useTheme } from 'react-native-paper'

interface ProductServiceCardProps {
	title: string
	description: string
	price: number
	priceFrequency: string
	category: string
	image: string
	onPress?: () => void
}

const ProductServiceCard: React.FC<ProductServiceCardProps> = ({
	title,
	description,
	price,
	priceFrequency,
	category,
	image,
	...props
}) => {
	const theme = useTheme()
	const styles = createStyles(theme)
	const { i18n, t } = useTranslation()

	const formattedPrice =
		new Intl.NumberFormat(i18n.language || 'en-US', {
			style: 'currency',
			currency: 'COP'
		}).format(price) +
		(priceFrequency ? ` / ${t('productService.' + priceFrequency)}` : '')

	return (
		<Card elevation={1} style={styles.card} {...props}>
			<Card.Cover source={{ uri: image }} />
			<Card.Content style={styles.cardContent}>
				<Text variant='titleLarge' style={styles.cardTitle}>
					{title}
				</Text>
				<Text variant='bodyMedium'>{description}</Text>
				<View style={styles.chipContainer}>
					<Chip textStyle={styles.chip} style={styles.chipBackground}>
						{t(`productService.${category}`)}
					</Chip>
					<Chip
						textStyle={styles.chipCost}
						style={styles.chipCostBackground}>
						{formattedPrice}
					</Chip>
				</View>
			</Card.Content>
		</Card>
	)
}

const createStyles = (theme: MD3Theme) =>
	StyleSheet.create({
		card: {
			backgroundColor: '#f2f2f2',
			paddingBottom: 10,
			width: '100%'
		},
		cardTitle: {
			fontWeight: 'bold'
		},
		cardContent: {
			justifyContent: 'center',
			marginTop: 10
		},
		chipContainer: {
			alignItems: 'flex-start',
			justifyContent: 'space-between',
			flexDirection: 'row',
			marginTop: 10
		},
		chipBackground: {
			backgroundColor: theme.colors.surfaceVariant
		},
		chip: {
			color: theme.colors.onPrimaryContainer,
			textTransform: 'capitalize'
		},
		chipCostBackground: {
			backgroundColor: theme.colors.tertiaryContainer
		},
		chipCost: {
			color: theme.colors.onTertiaryContainer
		}
	})

export default ProductServiceCard
