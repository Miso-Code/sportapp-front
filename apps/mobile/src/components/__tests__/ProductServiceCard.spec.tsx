import React from 'react'
import renderer, { ReactTestRenderer } from 'react-test-renderer'

import ProductServiceCard from '../ProductServiceCard'

jest.mock('dayjs')

describe('ProductServiceCard', () => {
	let component: ReactTestRenderer

	beforeEach(() => {
		component = renderer.create(<ProductServiceCard />)
	})

	afterEach(() => {
		component.unmount()
	})

	it('should render correctly', () => {
		expect(component.toJSON()).toMatchSnapshot()
	})

	it('should call onPress', () => {
		const onPress = jest.fn()
		component.update(<ProductServiceCard onPress={onPress} />)
		component.root.props.onPress()
		expect(onPress).toHaveBeenCalled()
	})
})
