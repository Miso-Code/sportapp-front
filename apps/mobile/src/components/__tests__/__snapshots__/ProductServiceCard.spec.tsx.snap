// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductServiceCard should render correctly 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "#f2f2f2",
      "borderRadius": 12,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
      "width": "100%",
    }
  }
  testID="card-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#f2f2f2",
        "borderRadius": 12,
        "elevation": 1,
        "flex": undefined,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
    testID="card-container"
  >
    <View
      style={
        [
          {
            "flexShrink": 1,
          },
          undefined,
        ]
      }
      testID="card"
    >
      <View
        style={
          [
            {
              "backgroundColor": "#eeeeee",
              "height": 195,
              "overflow": "hidden",
            },
            {
              "borderRadius": 12,
            },
            undefined,
          ]
        }
      >
        <Image
          accessibilityIgnoresInvertColors={true}
          borderRadiusStyles={{}}
          siblings={
            [
              "Card.Cover",
              "Card.Content",
            ]
          }
          source={
            {
              "uri": "test.jpg",
            }
          }
          style={
            [
              {
                "flex": 1,
                "height": undefined,
                "justifyContent": "flex-end",
                "padding": 16,
                "width": undefined,
              },
              {
                "borderRadius": 12,
              },
            ]
          }
        />
      </View>
      <View
        borderRadiusStyles={{}}
        style={
          [
            {
              "paddingHorizontal": 16,
            },
            {
              "paddingBottom": 16,
            },
            {
              "justifyContent": "center",
              "marginTop": 10,
              "paddingBottom": 10,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 22,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 28,
                },
                {
                  "fontWeight": "bold",
                },
              ],
            ]
          }
        >
          Test
        </Text>
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                },
                undefined,
              ],
            ]
          }
        >
          Test description
        </Text>
        <View
          style={
            {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 10,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "rgba(231, 224, 236, 1)",
                "borderRadius": 8,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="chip-container-outer-layer"
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "rgba(231, 224, 236, 1)",
                  "borderColor": "rgba(121, 116, 126, 1)",
                  "borderRadius": 8,
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "flex": undefined,
                  "flexDirection": "column",
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
              testID="chip-container"
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": true,
                    "expanded": undefined,
                    "selected": false,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    [
                      {
                        "borderRadius": 8,
                      },
                      {
                        "width": "100%",
                      },
                    ],
                  ]
                }
                testID="chip"
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "paddingLeft": 4,
                        "position": "relative",
                      },
                      {
                        "paddingLeft": 0,
                      },
                      {
                        "paddingRight": 0,
                      },
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      [
                        {
                          "textAlign": "left",
                        },
                        {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        [
                          {
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          [
                            {
                              "marginVertical": 6,
                              "textAlignVertical": "center",
                            },
                            {
                              "color": "rgba(29, 25, 43, 1)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            {
                              "marginLeft": 16,
                              "marginRight": 16,
                            },
                            {
                              "color": "rgba(33, 0, 93, 1)",
                              "textTransform": "capitalize",
                            },
                          ],
                        ],
                      ]
                    }
                  >
                    productService.Test category
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "rgba(255, 216, 228, 1)",
                "borderRadius": 8,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="chip-container-outer-layer"
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "rgba(255, 216, 228, 1)",
                  "borderColor": "rgba(121, 116, 126, 1)",
                  "borderRadius": 8,
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "flex": undefined,
                  "flexDirection": "column",
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
              testID="chip-container"
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": true,
                    "expanded": undefined,
                    "selected": false,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    [
                      {
                        "borderRadius": 8,
                      },
                      {
                        "width": "100%",
                      },
                    ],
                  ]
                }
                testID="chip"
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "paddingLeft": 4,
                        "position": "relative",
                      },
                      {
                        "paddingLeft": 0,
                      },
                      {
                        "paddingRight": 0,
                      },
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      [
                        {
                          "textAlign": "left",
                        },
                        {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        [
                          {
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          [
                            {
                              "marginVertical": 6,
                              "textAlignVertical": "center",
                            },
                            {
                              "color": "rgba(29, 25, 43, 1)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            {
                              "marginLeft": 16,
                              "marginRight": 16,
                            },
                            {
                              "color": "rgba(49, 17, 29, 1)",
                            },
                          ],
                        ],
                      ]
                    }
                  >
                    COP 100.00 / productService.monthly
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
